### ISSUE TITLE
Fstring and usage on TPU

### ISSUE THREAD

**Issue by vishnubanna** [tensorflow/recommenders#346]
Hi, 

So I have been trying to make use of the composite optimizer in the tf-recommenders package for an object detection project. There are issues that result from training the model on the TPU, the issues mostly stem from the usage of python f-string when raising a warning or an error. These prevent TPU compilation without:

```
tf.config.set_soft_device_placement(True)
```

An example of this usage from the Composite Optimizer apply gradients method is the following: 
```
raise ValueError(
    f"The set of variables handled by each optimizer should be "
    f"disjoint, but variable {v} is handled both "
    f"by {var_optimizer_dict[v.ref()]} and {optimizer}.")
```

This issue shows up in both tensorflow 2.4.0 and 2.5.0 
NOTE: removing the f-string from the optimizer etc. allows for full usage on the TPU.

**Comment by maciejkula** [comment]
Could you post the actual error you've seen?

You should probably be using `soft_device_placement` anyway, but I'm curious about the error.
